db/migrate/20180410071554_current_policy.rb

```diff
@@ -0,0 +1,58 @@
+Sequel.migration do
+  up do
+    alter_table :policy_versions do
+      set_column_type :created_at, :timestamptz
+      add_column :commited_at, :timestamptz,
+        index: true, # to quickly find the null
+        null: true # will be enforced by a trigger
+      add_constraint(:created_before_commit) { created_at <= commited_at }
+    end
+
+    execute """
+      CREATE OR REPLACE FUNCTION policy_versions_commit()
+        RETURNS trigger
+      LANGUAGE plpgsql AS $$
+        BEGIN
+          UPDATE policy_versions pv
+            SET commited_at = clock_timestamp()
+            WHERE commited_at IS NULL;
+          RETURN new;
+        END;
+      $$;
+
+      -- deferred constraint trigger will run on transaction commit
+      CREATE CONSTRAINT TRIGGER commit_current
+        AFTER INSERT ON policy_versions
+        INITIALLY DEFERRED
+        FOR EACH ROW
+        WHEN (NEW.commited_at IS NULL)
+        EXECUTE PROCEDURE policy_versions_commit();
+
+      -- if any version is current while creating new one, commit it
+      CREATE TRIGGER only_one_current
```

Comment:
When would we try and load a policy in a transaction when the previous one is still open? Should that be an error instead? I would rather forbid things that are weird and should never happen rather than try and handle them in a way that's hard to understand.

Reply:
Agreed. However, this will require refactoring some existing tests, and I wasn't sure we aren't doing it somewhere outside of tests, too.